src/Analysis/Engine/Impl/Infrastructure/PriorityProducerConsumer.cs

```diff
@@ -58,7 +58,17 @@ internal sealed class PriorityProducerConsumer<T> : IDisposable {
             _comparer = comparer ?? EqualityComparer<T>.Default;
         }
 
-        public void Dispose() => _disposeToken.TryMarkDisposed();
+        public void Dispose() {
+            if (_disposeToken.TryMarkDisposed()) {
+                lock (_syncObj) {
+                    Debug.Assert(_pendingTasks.Count == 0 || _firstAvailablePriority == _maxPriority);
+                    _pendingTasks.Clear();
```

Comment:
Should we cancel `pendingTcs`?

Reply:
They are canceled by `_disposeToken` itself:
https://github.com/Microsoft/python-language-server/blob/1d0c6244463ba740896ca78cc444f334b97454c0/src/Analysis/Engine/Impl/Infrastructure/PriorityProducerConsumer.cs#L135